front/pages/w/[wId]/builder/data-sources/[name]/edit-public-url.tsx

```diff
@@ -57,6 +59,11 @@ export const getServerSideProps = withDefaultUserAuthRequirements<{
     throw new Error(connectorRes.error.message);
   }
 
+  const dataSourceUsage = await getDataSourceUsage({
```

Comment:
Let's parallelize with the call above

Reply:
Please parallelize getDataSoruces and getDataSource above as well :pray:

Reply:
`getDataSourceUsage` requires the output of `getDataSource` as we need the `DataSource.id` to run `getDataSourceUsage`, right?

Reply:
Sorry I meant the Connector API call for parallelisation of this call

Reply:
Independently the two sequential calls at the top of the file can also be done concurrently